### DISCUSSION TITLE
Limiting response object properties

### DISCUSSION THREAD

**Discussion by vkuznecovas** [pocketbase/pocketbase#700] [Q&A] [answered]
Is there a way to omit some properties of the objects the API returns for a particular request? 

Here's the current response for `api/collections/kawa_pricing/records`:
```json
{
  "page": 1,
  "perPage": 30,
  "totalItems": 1357,
  "totalPages": 46,
  "items": [
    {
      "@collectionId": "8o9dnudhyalb7du",
      "@collectionName": "kawa_pricing",
      "created": "2022-10-02 18:38:45.555",
      "id": "1kelyccxjle7kg0",
      "planet": "Milliways",
      "price": 24.5,
      "ticker": "DW",
      "updated": "2022-10-02 18:38:45.555"
    },
    {
      "@collectionId": "8o9dnudhyalb7du",
      "@collectionName": "kawa_pricing",
      "created": "2022-10-02 18:38:45.555",
      "id": "ulauvzjlgwqjsyy",
      "planet": "Kiruna",
      "price": 40,
      "ticker": "RAT",
      "updated": "2022-10-02 18:38:45.555"
    },
    //... 
  ]
}
```

What I'd like to do is limit it to something this:
```json
{
  "page": 1,
  "perPage": 30,
  "totalItems": 1357,
  "totalPages": 46,
  "items": [
    {
      "planet": "Milliways",
      "price": 24.5,
      "ticker": "DW",
    },
    {
      "planet": "Kiruna",
      "price": 40,
      "ticker": "RAT",
    },
    //... 
  ]
}
```
Rest of the properties are of little benefit in this particular case and are just wasted bandwidth. 

Is there an existing way to achieve this?

If not I was thinking that maybe having a query param for this would be a nice addition, i.e. `api/collections/kawa_pricing/records?props=planet,price,ticker`.

Many thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
This was discussed in https://github.com/pocketbase/pocketbase/discussions/354.

Since this is just a minor optimization feature, for now it is left aside. I have some concerns about how the expanded relation fields will be listed but I'll revisit this and the hidden fields proposal in more details sometime in the near future.
